sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -339,6 +339,46 @@ public new AppendBlobClient WithVersion(string versionId)
                 ClientConfiguration);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AppendBlobClient"/>
+        /// class with an identical <see cref="Uri"/> source but the specified
+        /// <paramref name="customerProvidedKey"/>.
+        ///
+        /// </summary>
+        /// <param name="customerProvidedKey">The customer provided key.</param>
+        /// <returns>A new <see cref="AppendBlobClient"/> instance.</returns>
+        /// <remarks>
+        /// Pass null to remove the customer provide key in the returned <see cref="AppendBlobClient"/>.
+        /// </remarks>
+        public new AppendBlobClient WithCustomerProvidedKey(CustomerProvidedKey? customerProvidedKey)
```

Comment:
why `public new` ? do we need this on base client?

Reply:
Very good question.  I was pattern matching the existing With()'s.  I believe `new` is needed to override the base class implementation.

Reply:
nvm. we do that with other `with` methods.

Reply:
@kasobol-msft, I've been thinking about it more, and I think `new` is needed because `AppendBlobClient.WithCustomerProvidedKey()` has a different return type (AppendBlobClient) from the base implementation (BaseBlobClient)